Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+  {
+    var isEnabled = DropDownMenu.GetItemEnabled(itemInfo, selectionCount);
+
+    var item = document.createElement("li");
+
+    var className = DropDownMenu._itemClassName;
+    if (!isEnabled)
+      className = DropDownMenu._itemDisabledClassName;
+
+    item.className = className;
+    item.setAttribute('role', 'none');
+
+    var anchor = document.createElement("a");
+    anchor.setAttribute('role', 'menuitem');
+    anchor.setAttribute('tabindex', '-1');
+    if (isEnabled)
+    {
+      anchor.setAttribute('href', '#');
+    }
+    else
+    {
+      anchor.setAttribute('aria-disabled', 'true');
+    }
+    // TODO: missing nullcheck?
+    $(anchor).bind('click', DropDownMenu._itemClickHandler!);
+
+    item.appendChild(anchor);
+    if (isEnabled && itemInfo.Href != null)
+    {
+      var isJavaScript = itemInfo.Href.toLowerCase().indexOf('javascript:') >= 0;
+      if (isJavaScript)
+      {
+        anchor.setAttribute('javascript', itemInfo.Href);
+      }
+      else
+      {
+        var href = itemInfo.Href;
+        var target;
+        if (itemInfo.Target != null && itemInfo.Target.length > 0)
+          target = itemInfo.Target;
+        else
+          target = '_self';
+        anchor.setAttribute('javascript', 'window.open (\'' + href + '\', \'' + target + '\');');
+      }
+    }
+
+    if (itemInfo.Icon != null)
+    {
+      var icon = document.createElement('img');
+      if (isEnabled || itemInfo.IconDisabled == null)
+        icon.src = itemInfo.Icon;
+      else
+        icon.src = itemInfo.IconDisabled;
+
+      icon.className = DropDownMenu._itemIconClassName;
+      icon.style.verticalAlign = 'middle';
+      icon.alt = '';
+      icon.setAttribute('aria-hidden', 'true');
+      anchor.appendChild(icon);
+    }
+    else
+    {
+      var iconPlaceholder = document.createElement('span');
+      iconPlaceholder.className = DropDownMenu._itemIconClassName;
+      anchor.appendChild(iconPlaceholder);
+    }
+
+    if (itemInfo.DiagnosticMetadata)
+    {
+      // Do not render empty diagnostic metadata attributes
+      $.each(itemInfo.DiagnosticMetadata, function (key: string, value: string | boolean)
+      {
+        if (value === "" || value === null)
+        {
+          delete itemInfo.DiagnosticMetadata![key];
+        }
+      });
+
+      $ (item).attr (itemInfo.DiagnosticMetadata);
+    }
+
+    if (itemInfo.DiagnosticMetadataForCommand)
+    {
+      // Do not render empty diagnostic metadata attributes
+      $.each(itemInfo.DiagnosticMetadataForCommand,
+        function (key, value) {
+          if (value === "" || value === null) {
+            delete itemInfo.DiagnosticMetadataForCommand![key];
+          }
+        });
+
+      itemInfo.DiagnosticMetadataForCommand['data-is-disabled'] = isEnabled ? 'false' : 'true';
+
+      $(anchor).attr(itemInfo.DiagnosticMetadataForCommand);
+    }
+
+    var span = document.createElement('span');
+    span.innerHTML = itemInfo.Text!;
```

Comment:
What happens if Text is null?

Reply:
Nothing.

Reply:
okay. am wondering if we should still use `|| ''` here to make this more explicit that a) null is properly handled and b) we intend that it's empty in this case.